File: front_end_web/front_alt/src/clients/pedidoClient.js
import { connRefactor } from "./req.conf"


const saveOrder = (pedidos) => {
    let list = []
        pedidos.pratos.forEach(element => {
            list.push({id: element.id, quantidade: element.quantidade})
        });
        let body = JSON.stringify({
            id_cliente: 1,
            pratos: list
        })
        
    connRefactor.post('/servico/pedido-confirmado', body).then((r) => { return r.data;})
}

export default {saveOrder}

File: restaurante_es1/lib/client/pedidoClient.dart
import 'package:restaurante_es1/api/api.dart';

import 'package:http/http.dart' as http;
import 'package:restaurante_es1/model/Pedido.dart';
import 'dart:convert' show json;

class PedidosClient {
  String _urlBase = API.urlBase;
  Map<String, String> _headers = {
    "Content-Type": "application/json",
    "Access-Control_Allow_Origin": "*",
    "Content-Length": "x",
  };

  Future<void> saveOrder(
      int id, double valueTotal, List<Pedido> pedidos) async {
    var list = [];
    for (var item in pedidos) {
      list.add({"id": item.id, "quantidade": item.quantidade});
    }
    String body = json.encode({"id_cliente": 1, "pratos": list});
    print(body);

    var res = await http.post(
      Uri.parse('$_urlBase/servico/pedido-confirmado'),
      headers: _headers,
      body: body,
    );
    print(res.body);

    return;
  }
}
